### PR TITLE
Choose Team before joining session

### PR DESCRIPTION
**Pull request by bruderol**
- MVP of Client to choose a team
- fixed server to not place player and broadcast session joined events too early

### CODE REVIEW

File: client/js/gameSetup/chooseTeam.jsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+import React from 'react';
+import ExistingSessions from './existingSessions.jsx';
```
Comment (r3dDoX) [inline client/js/gameSetup/chooseTeam.jsx]: this import isn't used

File: client/js/gameSetup/chooseTeam.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import GameSetupStore from './gameSetupStore.js';
+import JassActions from '../jassActions.js';
+import {SessionType} from '../../../shared/session/sessionType.js';
+
+export default React.createClass({
+
+    joinTeam1: function() {
```
Comment (r3dDoX) [inline client/js/gameSetup/chooseTeam.jsx]: it would be nicer to handle this directly in a lambda on button click

File: client/js/jassActions.js [hunk 3]
```diff
@@ -32,28 +32,50 @@ export default {
...
         });
     },
 
-    joinExistingSessionAsSpectator: (sessionChoice) => {
+    joinExistingSessionAsSpectator: (sessionName) => {
         JassAppDispatcher.handleViewAction({
             actionType: JassAppConstants.CHOOSE_EXISTING_SESSION_SPECTATOR,
-            data: sessionChoice
+            data: {
+                sessionName: sessionName
```
Comment (r3dDoX) [inline client/js/jassActions.js]: can be simplified to data: {
  sessionName
}

File: server/session/singleGameSession.js [hunk 4]
```diff
@@ -86,30 +82,44 @@ function insertPlayer(session, player) {
...
 /**
  * Only broadcast the session joined event for all players that have been placed in the right table position,
  * for lastPlayerJoined and players behind.
  * @param session the session to broadcast for
  * @param lastPlayerJoined the last player joined
  */
-function broadcastSessionJoinedForCorrectPlacedPlayers(session, lastPlayerJoined) {
+function registerClientAndBroadcastSessionJoinedForCorrectPlacedPlayers(session, webSocket, lastPlayerJoined) {
+
+    // Prpare player registration finalization (tobe called later, when player is placed on table)
```
Comment (r3dDoX) [inline server/session/singleGameSession.js]: typo on "Prepare"

File: client/js/gameSetup/chooseTeam.jsx (hunk 1 above)
Comment (bruderol) [inline client/js/gameSetup/chooseTeam.jsx]: allready fixed

File: client/js/gameSetup/chooseTeam.jsx (hunk 2 above)
Comment (bruderol) [inline client/js/gameSetup/chooseTeam.jsx]: Yes, I allready started to improve on this

File: server/session/singleGameSession.js (hunk 4 above)
Comment (bruderol) [inline server/session/singleGameSession.js]: Thanks for correction

**Pull request merged**